### ISSUE TITLE
Observability: Envoy Proxy

### ISSUE THREAD

**Issue by arkodg** [envoyproxy/gateway#699]
*Description*:
This epic tracks the design and implementation for surfacing `logs`, `metrics` and `traces` generated by Envoy Proxy that are relevant to the user personas

**Comment by arkodg** [comment]
related PR for metrics https://github.com/envoyproxy/gateway/pull/581/files

**Comment by arkodg** [comment]
related PR for traces https://github.com/envoyproxy/gateway/pull/697

**Comment by zirain** [comment]
best practices form prometheus: https://prometheus.io/docs/practices/naming/#metric-and-label-naming

**Comment by zirain** [comment]
accesslog: I think should focus on File(send /dev/stdout) and OpenTelemetry(I believe it's the future)
traces: only support OpenTelemetry is an options

**Comment by danehans** [comment]
@Xunzhuo unless you're actively working on this issue, can you unassign yourself and add the help-wanted label?

**Comment by arkodg** [comment]
should we configure envoy so it sends OLTP format to otel-collector and the user configures EG API to specify the  Otel-collector endpoint (user (or docs) brings their otel-collector)
There seems to be support for [access log](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/access_loggers/open_telemetry/v3/logs_service.proto) [and traces](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/trace/v3/opentelemetry.proto.html) and see a [PR for metrics support](https://github.com/envoyproxy/envoy/pull/26620)

**Comment by arkodg** [comment]
are there any existing grafana dashboards we can reuse to showcase metrics ?

**Comment by zirain** [comment]
I made two proposals before:
1. via EnvoyProxy https://github.com/envoyproxy/gateway/pull/697
2. via PolicyAttachment https://github.com/envoyproxy/gateway/pull/1121

so I think this time just add configuration in EnvoyGateway?

cc @arkodg @AliceProxy @kflynn @Xunzhuo

**Comment by arkodg** [comment]
@zirain I vote for defining Data Plane Observability/Telemetry within the `EnvoyProxy` resource .  Your PRs gain should gain more traction and reviews in the coming weeks, now that Observability is the theme for the next release.
Also request to add a design doc highlighting the - why, what and how.
Tackling one signal at a time (access logging first, then the rest) should also speed up the review process

**Comment by zirain** [comment]
ok, I will move forward https://github.com/envoyproxy/gateway/pull/697

**Comment by arkodg** [comment]
lets keep this issue open to track completion of docs
moving this to v0.5

**Comment by LanceEa** [comment]
@arkodg @zirain - I was testing some of this out. I just wanted to record a couple of findings that might help with documentation.

First, the OTEL Stats Sink didn't land until Envoy 1.27 so if trying to test EG against an older version of Envoy then it will fail. This kind of obvious and with the EG policy but just mention it in case anyone else had pinned an older version of Envoy and expected it to work. Might be a good call out in the docs, thoughts? happy to throw up a docs PR.

Second,  I'm testing `metrics.prometheus` using the PrometheusOperator and ran into a few road bumps. By default, no `Service` exposes the 19001 port so that means `ServiceMonitor` will not work. I tried `PodMonitor` to scrape the pods directly but that didn't like the fact that the Envoy Proxy Deployment doesn't expose the `19001` port (_only the gateway.listener ports_) for scraping metrics directly from the EnvoyProxy pods.

I ended up adding my own `Service` that exposed the `19001` port to the cluster and used a `ServiceMonitor` and it started to work.

So, I guess my questions are:
1. Am I completely missing something or a document already outlining this 😄 (_totally possible_)?
2. Do we want to consider this a bug and address it before 0.5.0 lands? I would think that the InfraManager should inject the port into the Deployment and probably expose a second `Service` of type `ClusterIP`.
3. If we do not have time to address it for 0.5.0 and there are not docs yet then can we add a little snippet about this temporary work-around?
